### DISCUSSION TITLE
Workaround for error with Link's 'as' and 'href' incompatible values

### DISCUSSION THREAD

**Discussion by knapcp3** [vercel/next.js#21340] [Help]
I created following page in my Next.js project: `src/pages/brand1/[[...subpages]]`. I'm using Incremental Static Regeneration, I have `.next/server/pages/brand1.json` file after building next.js project. In my component I want to create a `Link` which when clicked, in address bar displays URL with empty path (for example `page-domain.com`) and renders the page from `/brand1` route (which uses `brand1.json` file). I tried various solutions to achieve this, for example:

```
<Link
  href={{
    pathname: '/brand1',
  }}
  as={'/'}
  shallow
>
  link
</Link>
```
But I get the error in browser saying that `Error: The provided 'as' value (/) is incompatible with the 'href' value (/brand1/[[...subpages]])`

I have following solution which works for urls with not empty subpages (for example `page-domain.com/menu`), but it doesn't work for the root page as I described earlier:
```
<Link
  href={{
    pathname: '/brand1/[[...subpages]]',
    query: { subpages: 'menu' },
  }}
  as={'/menu'}
  shallow
>
  link
</Link>
```

My question is how to create link which can display URL with empty path like `page-domain.com` and gets the page from `/brand1` route?

**Comment by damjtoh** [comment]
did you find a solution to this?

**Reply by knapcp3** [reply]
Nope, I couldn't find solution to this anywhere 😭
